fetch.sh

```diff
@@ -31,44 +35,117 @@ esac
-        https://kojipkgs.fedoraproject.org//packages/kernel/"$KERNEL_MAJOR_MINOR_PATCH"/"$KERNEL_RELEASE"/"$ARCH"/kernel-core-"$kernel_version".rpm \
         https://kojipkgs.fedoraproject.org//packages/kernel/"$KERNEL_MAJOR_MINOR_PATCH"/"$KERNEL_RELEASE"/"$ARCH"/kernel-modules-"$kernel_version".rpm \
         https://kojipkgs.fedoraproject.org//packages/kernel/"$KERNEL_MAJOR_MINOR_PATCH"/"$KERNEL_RELEASE"/"$ARCH"/kernel-modules-core-"$kernel_version".rpm \
         https://kojipkgs.fedoraproject.org//packages/kernel/"$KERNEL_MAJOR_MINOR_PATCH"/"$KERNEL_RELEASE"/"$ARCH"/kernel-modules-extra-"$kernel_version".rpm \
         https://kojipkgs.fedoraproject.org//packages/kernel/"$KERNEL_MAJOR_MINOR_PATCH"/"$KERNEL_RELEASE"/"$ARCH"/kernel-devel-"$kernel_version".rpm \
         https://kojipkgs.fedoraproject.org//packages/kernel/"$KERNEL_MAJOR_MINOR_PATCH"/"$KERNEL_RELEASE"/"$ARCH"/kernel-devel-matched-"$kernel_version".rpm \
         https://kojipkgs.fedoraproject.org//packages/kernel/"$KERNEL_MAJOR_MINOR_PATCH"/"$KERNEL_RELEASE"/"$ARCH"/kernel-uki-virt-"$kernel_version".rpm
+
 fi
 
 if [[ "${kernel_flavor}" =~ fsync ]]; then
     dnf download -y \
         kernel-headers-"${kernel_version}"
 fi
 
+if [[ ! -s /tmp/certs/private_key.priv ]]; then
+    echo "WARNING: Using test signing key."
+    cp /tmp/certs/private_key.priv{.test,}
+    cp /tmp/certs/public_key.der{.test,}
+fi
+
+PUBLIC_KEY_PATH="/etc/pki/kernel/public/public_key.crt"
+PRIVATE_KEY_PATH="/etc/pki/kernel/private/private_key.priv"
+
+openssl x509 -in /tmp/certs/public_key.der -out /tmp/certs/public_key.crt
+
+install -Dm644 /tmp/certs/public_key.crt "$PUBLIC_KEY_PATH"
+install -Dm644 /tmp/certs/private_key.priv "$PRIVATE_KEY_PATH"
+
+if [[ "${kernel_flavor}" =~ asus|fsync ]]; then
+    dnf install -y \
+        /kernel-"$kernel_version".rpm \
+        /kernel-modules-"$kernel_version".rpm \
+        /kernel-modules-core-"$kernel_version".rpm \
+        /kernel-modules-extra-"$kernel_version".rpm \
+        kernel-core-"${kernel_version}"
+elif [[ "${kernel_version}" =~ surface ]]; then
+    dnf install -y \
+        /kernel-surface-"$kernel_version".rpm \
+        /kernel-surface-modules-"$kernel_version".rpm \
+        /kernel-surface-modules-core-"$kernel_version".rpm \
+        /kernel-surface-modules-extra-"$kernel_version".rpm \
+        kernel-surface-core-"${kernel_version}"
+else
+    dnf install -y \
+        /kernel-"$kernel_version".rpm \
+        /kernel-modules-"$kernel_version".rpm \
+        /kernel-modules-core-"$kernel_version".rpm \
+        /kernel-modules-extra-"$kernel_version".rpm \
+        https://kojipkgs.fedoraproject.org//packages/kernel/"$KERNEL_MAJOR_MINOR_PATCH"/"$KERNEL_RELEASE"/"$ARCH"/kernel-core-"$kernel_version".rpm
+fi
+
+# Strip Signatures from non-fedora Kernels
+if [[ ${kernel_flavor} =~ asus|surface|fsync ]]; then
```

Comment:
I'd suggest this match what we know we don't want to strip:

```
if [[ ${kernel_flavor} =~ main|coreos ]]; then
    echo "will not strip Fedora signatures from ${kernel_flavor} kernels"
else
```

This way we'll always be safe on Fedora kernels, and anything else automatically gets stripped, don't have to manage the list.